### ISSUE TITLE
“Always Open in This Container” doesn’t work on Firefox startup

### ISSUE THREAD

**Issue by ArchangeGabriel** [mozilla/testpilot-containers#474]
Test case:
- Firefox is not launched
- I open a link from elsewhere (e.g. a GitHub email notifications in Thunderbird)
- Firefox launches, opening the link in the default containers instead of the specified one if any.

**Comment by groovecoder** [comment]
Good catch ... I'm not sure we can fix this in Test Pilot add-on - I imagine we can't jump into the network request that early.

**Comment by jonathanKingston** [comment]
@andymckay is this something your team would be interested in to solving / having a bug for. Basically the issue is on start up our extension doesn't capture the webrequest before it is loaded.

**Comment by andymckay** [comment]
Sure, not a simple bug to solve, there might be one already. There's a conflict between keeping startup times fast - not blocking on WebExtensions in startup - and this sort of request.

**Comment by geeknik** [comment]
I find that the “Always Open in This Container” doesn't work period. I set up a container, open up a site I don't like in it, click the option to “Always Open in This Container” and then close the container. When I visit the site I don't like, it opens in a normal tab and not the container I configured.

**Comment by jonathanKingston** [comment]
@geeknik can you open a separate issue with clear Steps To Reproduce? That sounds different and not related to this.

@andymckay yeah I am guessing it won't be something simple to fix at all but I would rather have it logged somewhere as a limitation of that API.

**Comment by kjozwiak** [comment]
@jonathanKingston, @geeknik ended up creating #484 but I couldn't reproduce the issue after a few hours of attempts on different machines/platforms/add-ons. I'm not sure if it was his particular setup but if this was more wide spread, we would definitely be seeing more bugs.

**Comment by kjozwiak** [comment]
This also affects `When Nightly starts: Show your home page`:

STR:

* install fx and the [containers experiment](https://testpilot.firefox.com/experiments/containers) via [test pilot](https://testpilot.firefox.com/) (either 2.2.0 or 2.3.0)
* open a personal container and visit facebook.com
* right click on facebook.com and select `Always Open in This Container`
* close the personal tab and visit `about:preferences`
* select `When Nightly starts: Show your homepage` and under `Home Page:` use facebook.com
* restart fx

**Comment by bakulf** [comment]
Maybe it's not the cleanest way to fix this issue, but when container addon starts, it can check if there are already opened tab and if they are in the wrong container, we can reload them correctly...

**Comment by jonathanKingston** [comment]
@bakulf I think that would create as many issues as it would solve personally.

**Comment by grahamperrin** [comment]
Please tell whether symptoms are reproducible with the latest extension added to Firefox `88`.

See https://github.com/mozilla/multi-account-containers/issues/1719#issuecomment-825747906
